esx_service/utils/vmdk_utils.py

```diff
@@ -100,8 +100,11 @@ def get_datastores():
     'url-name' is the last element of datastore URL (e.g. 'vsan:572904f8c031435f-3513e0db551fcc82')
     'dockvol-path; is a full path to 'dockvols' folder on datastore
     """
-    if datastores == None:
-        init_datastoreCache()
+    # if datastores == None:
+    #     init_datastoreCache()
```

Comment:
This change is needed. Think about this case:
step 1: assume we have datastore named "datastore1", which is the default_datastore, we create a volume "vol1"
step 2: rename "datastore1" to "datastore 1" (including space in the name)
step3: create another vol "vol2", which will fail.  

It will hit the error same as #818 . The root cause is "vmdk_utils.get_datastores" returns cached value which is outdated.